.github/workflows/ci.yaml

```diff
@@ -12,12 +12,13 @@ jobs:
       - name: install
         uses: actions/setup-go@v5
         with:
-          go-version: 1.21.x
+          go-version: 1.22.x
+          cache: false
```

Comment:
Cache isn't used as no `go.sum` present, but explicitly disable to avoid warning.

Reply:
can/should this be a comment in the yaml itself?

Reply:
I've changed this to use `cache-dependent-path` in the setup-go action which looks to be much better at speeding up the workflow (I think from caching build artifacts). For example:
- cache hit using setup-go: https://github.com/bufbuild/registry-proto/actions/runs/9357564225/job/25757623058
- cache hit using the custom cache step: https://github.com/bufbuild/registry-proto/actions/runs/9357179796/job/25756335481

Results: 6s for setup-go vs 41s for toolkit cache.